subsys/bluetooth/shell/Makefile

```diff
@@ -2,3 +2,7 @@ obj-y += bt.o
 obj-y += gatt.o
 obj-$(CONFIG_BLUETOOTH_CONTROLLER) += ll.o
 obj-$(CONFIG_BLUETOOTH_CONTROLLER) += ticker.o
+
+ifeq ($(CONFIG_SOC_FLASH_NRF5),y)
+obj-$(CONFIG_SOC_FLASH_NRF5) += flash.o
+endif
```

Comment:
The ifeq-endif pair is unnecessary. That's the main convenience with assigning to obj-$(CONFIG_FOO).